pkg/purge_osd/purge_osd.go

```diff
@@ -0,0 +1,33 @@
+package purgeosd
+
+import (
+	"context"
+	"fmt"
+
+	cephcommands "github.com/red-hat-storage/odf-cli/pkg/ceph"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	"github.com/rook/kubectl-rook-ceph/pkg/rook"
+)
+
+func PurgeOsd(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace, cephClusterNamespace, osdID, forceflagValue string) {
+	var answer string
+	logging.Warning("Are you sure you want to purge osd ID %s? If so, enter 'yes'\n", osdID)
+	fmt.Scanf("%s", &answer)
```

Comment:
As much as possible let's change the behavior of the commands in the upstream tool. In this case, we can improve the purge-osd implementation upstream so that it prompts to really purge the OSD. It could also detect first if the OSD can safely be removed without the prompt, before the prompt is shown. Then we don't need to implement this prompt in the downstream tool.

Reply:
In upstream when we trigger command `ceph osd remove ...` the rook command to remove the osd we first mark osd out and the if osd is `safe-to-destroy` here https://github.com/rook/rook/blob/master/pkg/daemon/ceph/osd/remove.go#L90 so in that case we should not need this to check again.

@travisn  The reason I added the prompt and check pg status as these were requested from support team to add these pre check.

Reply:
Why not just make these improvements in the upstream tool? I don't see why it's specific to downstream.

The whole purpose of the `safe-to-destroy` is to check if the PGs would allow removing the OSD safely. If the PGs aren't clean enough to allow the OSD to be safely removed, then we wouldn't remove it without the force flag. So I don't see why the additional PG check helps. Perhaps the more important question is whether we allow the `--force` flag to proceed, or if we would require prompting to type something like `yes-really-force-osd-removal`.

Reply:
>So I don't see why the additional PG check helps. Perhaps the more important question is whether we allow the --force flag to proceed, or if we would require prompting to type something like yes-really-force-osd-removal.

This is valid question I guess, I'll try get answer from Bipin.